velox/dwio/parquet/tests/writer/ParquetWriterTest.cpp

```diff
@@ -146,6 +147,59 @@ TEST_F(ParquetWriterTest, compression) {
   assertReadWithReaderAndExpected(schema, *rowReader, data, *leafPool_);
 };
 
+TEST_F(ParquetWriterTest, datapageVersion) {
+  auto schema = ROW({"c0"}, {INTEGER()});
+  const int64_t kRows = 1;
+  const auto data = makeRowVector({
+      makeFlatVector<int32_t>(kRows, [](auto row) { return 987; }),
+  });
+
+  // Set parquet datapage version and write data - then read to ensure the
+  // property took effect.
+  const auto testDataPageVersion = [&](bool useParquetDataPageV2) {
+    // Create an in-memory writer.
+    auto sink = std::make_unique<MemorySink>(
+        200 * 1024 * 1024,
+        dwio::common::FileSink::Options{.pool = leafPool_.get()});
+    auto sinkPtr = sink.get();
+    parquet::WriterOptions writerOptions;
```

Comment:
@svm1 can we use `processConfigs` to ensure we test the config code path? Thanks.

Reply:
Great point, thanks. When I originally wrote this test, config processing logic was not yet decoupled from HiveDataSink, so manually editing WriterOptions seemed like the best I could do.

Rewrote the test to simulate modifying datapage flag via both config file and session property, to test the actual config code path.